### PR TITLE
app-crypt/trousers: Cleanup old versions, POSIXify init script

### PR DESCRIPTION
**Pull request by salahcoronya**
The init script for tcsd contains a lot of machinery that isn't needed anymore, some if which contained bashisms. So cleaned up the initscript to make it more modern.

### CODE REVIEW

File: app-crypt/trousers/files/tcsd.initd [hunk 1]
```diff
@@ -1,38 +1,12 @@
...
-		for mod in ${TPM_MODULES} ; do
-			lsmod | grep -q "^${mod}\b" \
-				|| modprobe ${mod} &>/dev/null \
-				|| ewarn "Failed to load module ${mod}"
-		done
-		# Should we sleep or something to wait for device creation?
-	fi
-	if [ ! -c /dev/tpm ] && [ ! -c /dev/tpm0 ] ; then
-		eerror "No TPM device found!"
-		return 1
```
Comment (juippis) [inline app-crypt/trousers/files/tcsd.initd:22]: At least this was useful, now it just says "can't start the service" and leaves me guessing what's wrong :(

File: app-crypt/trousers/files/tcsd.initd [hunk 2]
```diff
@@ -1,38 +1,12 @@
 #!/sbin/openrc-run
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2022 Gentoo Foundation
```
Comment (juippis) [inline app-crypt/trousers/files/tcsd.initd:2]:
```suggestion
# Copyright 1999-2022 Gentoo Authors
```

File: app-crypt/trousers/files/tcsd.initd (hunk 1 above)
Comment (salahcoronya) [reply app-crypt/trousers/files/tcsd.initd]: I put that back in

File: app-crypt/trousers/files/tcsd.initd (hunk 2 above)
Comment (salahcoronya) [reply app-crypt/trousers/files/tcsd.initd]: Done

**Review by juippis — APPROVED** [review]
Thanks!

**Pull request closed without merging**